tests/trace/test_dataset.py

```diff
@@ -141,3 +141,32 @@ def test_dataset_caching(client):
 
     with raise_on_captured_errors():
         assert len(ds2) == 200
+
+
+def test_add_rows(client):
+    ds = weave.Dataset(name="test", rows=[{"a": i} for i in range(10)])
+    ref = weave.publish(ds)
+
+    ds = ref.get()
+    ds2 = ds.add_rows([{"a": 10}])
+
+    assert len(ds2) == 11
+    assert ds2.rows[10]["a"] == 10
+
+    ds3 = ds2.add_rows([{"a": 11}, {"a": 12}, {"a": 13}])
+    assert len(ds3) == 14
+    assert ds3.rows[12]["a"] == 12
+    assert ds3.rows[11]["a"] == 11
+    assert ds3.rows[10]["a"] == 10
```

Comment:
You should additionally publish `ds2`, get it back as `ds3`, then do your assertions.

Reply:
The add rows function publishes them

Reply:
In that case I think you can still add a publish to the test and note that it won't do anything.

The case I'm trying to cover is where the user:
1. Adds rows to a ref-gotten dataset; and
2. Publishes and tries to get that dataset back

Reply:
I added a check that calling weave.publish(...).get() doesn't change the rows

Reply:
thanks!